### PR TITLE
Cleaning, fixing, tweaking up chemistry_reagents.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Cleaning up chemistry reagents. Standarizing many mob_related reagent procs to /living (previously fluctuating between /living and  /carbon with potential runtimes). Adding parent calls to the end of subtypes' on_mob_life() where previously missing or doing it earlier. Tweaking some effects, such as removing RNG inconsistency of `prob(20) adjustToxLoss(1)` with `adjustToxLoss(0.2)`. Made xeno larval growth toxin a subtype of /toxin. Made subtypes `reaction_mob()` early call the parent where opportune.
Removed some `!alien` tag checks, to make some medicines work as intended on slighty-less-human mobs.

## Why It's Good For The Game
Improving and fixing and tweaking the module. I commited handfuls of mistakes when first porting it.

## Changelog
:cl:
tweak: Tweaked how some reagents work. removed a little RNG, made xeno larval growth serum into a subtype of toxin (with no tox pwr).
fix: Certain medicines (such as tricordrazine should now) properly work on slighty-less-human mobs such as mothpeople.
code: standarizing reagents procs from working on /carbon level to /living level a little more.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry_reagents/food.dm [hunk 1]
```diff
@@ -158,61 +160,68 @@
...
-					if(prob(10))
-						victim.emote("scream")
-				to_chat(victim, "<span class='danger'>You're sprayed directly in the eyes with pepperspray!</span>")
-				victim.blur_eyes(25)
-				victim.blind_eyes(10)
-				victim.Stun(5)
-				victim.KnockDown(5)
-				//victim.KnockOut(10)
-				//victim.drop_held_item()
+/datum/reagent/consumable/capsaicin/condensed/reaction_mob(mob/living/L, method = TOUCH, volume, alien, show_message = TRUE, touch_protection = FALSE)
```
Comment (Ghommie) [inline code/modules/reagents/chemistry_reagents/food.dm]: This one will have to be written over again with a future implementation of `is_mouth_covered() `and `are_eyes_covered()` anyway, so didn't do much than removing indention.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -34,93 +34,93 @@
...
+/datum/reagent/proc/on_mob_life(mob/living/L, alien)
+	purge(L)
 	current_cycle++
-	holder.remove_reagent(id, custom_metabolism * M.metabolism_efficiency) //By default it slowly disappears.
+	holder.remove_reagent(id, custom_metabolism * L.metabolism_efficiency) //By default it slowly disappears.
 	return TRUE
 
 // Called when this reagent is first added to a mob
-/datum/reagent/proc/on_mob_add(mob/living/L)
+/datum/reagent/proc/on_mob_add(mob/living/L, alien)
```
Comment (DominikPanic) [inline code/modules/reagents/Chemistry-Reagents.dm]: Why is it named alien?

File: code/modules/reagents/chemistry_reagents/toxin.dm [hunk 3]
```diff
@@ -242,51 +240,48 @@
...
-			if(H.dna)
-				if(H.species.species_flags & IS_PLANT) //plantmen take a LOT of damage
-					H.adjustToxLoss(10)
+/datum/reagent/toxin/plantbgone/reaction_mob(mob/living/L, method = TOUCH, volume, alien, show_message = TRUE, touch_protection = FALSE)
+	. = ..()
+	if(!. || !ishuman(L))
+		return
+	var/mob/living/carbon/human/H = L
+	if(H.dna)
+		if(H.species.species_flags & IS_PLANT) //plantmen take a LOT of damage
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/toxin.dm]: Could combine those two

File: code/modules/reagents/chemistry_reagents/other.dm [hunk 4]
```diff
@@ -9,32 +9,26 @@
...
 	var/list/L = list()
-	if(self.data["blood_DNA"])
-		L = list(self.data["blood_DNA"] = self.data["blood_type"])
-
-	T.add_blood(L , self.color)
+	if(data["blood_DNA"])
+		L = list(data["blood_DNA"] = data["blood_type"])
+	T.add_blood(L , color)
 
 /datum/reagent/blood/yaut_blood
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/other.dm]: Remove this entirely

File: code/modules/reagents/chemistry_reagents/other.dm [hunk 5]
```diff
@@ -771,51 +754,29 @@
...
-	overdose_crit_threshold = REAGENTS_OVERDOSE_CRITICAL
-	scannable = TRUE
+		L.KnockDown(1)
+	return ..()
 
 
-/datum/reagent/xeno_growthtoxin/overdose_process(mob/living/M)
-	M.adjustOxyLoss(3)
-	M.Jitter(4) //Lets Xenos know they're ODing and should probably stop.
+/datum/reagent/xeno_neurotoxin/overdose_process(mob/living/L, alien)
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/other.dm]: Should also move this into the toxin.dm

File: code/modules/reagents/chemistry_reagents/medical.dm [hunk 6]
```diff
@@ -125,41 +128,51 @@
...
-	if(method in list(TOUCH, VAPOR, PATCH))
-		M.germ_level -= min(volume*20, M.germ_level)
-		if((M.getFireLoss() > 30 || M.getBruteLoss() > 30) && prob(10)) // >Spraying space bleach on open wounds
-			to_chat(M, "<span class='warning'>Your open wounds feel like they're on fire!</span>")
-			M.emote(pick("scream","pain","moan"))
-			M.reagent_shock_modifier -= PAIN_REDUCTION_MEDIUM
-
-/datum/reagent/medicine/sterilizine/reaction_obj(var/obj/O, var/volume)
+/datum/reagent/medicine/sterilizine/reaction_mob(mob/living/L, method = TOUCH, volume, alien, show_message = TRUE, touch_protection = FALSE)
+	if(!istype(L))
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/medical.dm]: Will this ever happen?

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 7]
```diff
@@ -34,93 +34,93 @@
 	. = ..()
 	holder = null
 
-/datum/reagent/proc/reaction_mob(var/mob/living/M, method=TOUCH, reac_volume, show_message = 1, touch_protection = 0)
-	if(!istype(M))
-		return 0
+/datum/reagent/proc/reaction_mob(mob/living/L, method = TOUCH, volume, alien, show_message = TRUE, touch_protection = FALSE)
+	if(!istype(L))
```
Comment (DominikPanic) [inline code/modules/reagents/Chemistry-Reagents.dm]: Same, will this ever be true?

File: code/modules/reagents/chemistry_reagents/drink.dm [hunk 8]
```diff
@@ -221,31 +218,31 @@
...
 	if(adj_temp > 0 && holder.has_reagent("frostoil"))
 		holder.remove_reagent("frostoil", 5)
 	return ..()
 
-/datum/reagent/consumable/drink/coffee/overdose_process(mob/living/M, alien)
-	M.Jitter(5)
-	if(ishuman(M))
+/datum/reagent/consumable/drink/coffee/overdose_process(mob/living/L, alien)
+	L.Jitter(5)
+	if(ishuman(L))
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/drink.dm]: combine these two with prob5 first for efficiency

File: code/modules/reagents/chemistry_reagents/drink.dm [hunk 9]
```diff
@@ -221,31 +218,31 @@
...
-	if(ishuman(M))
+			L.emote(pick("twitch", "blink_r", "shiver"))
+
+/datum/reagent/consumable/drink/coffee/overdose_crit_process(mob/living/L, alien)
+	L.apply_damage(0.2, TOX)
+	L.Jitter(5)
+	if(prob(5) && L.stat != UNCONSCIOUS)
+		to_chat(L, "<span class='warning'>You spasm and pass out!</span>")
+		L.KnockOut(5)
+	if(ishuman(L))
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/drink.dm]: Same here

File: code/modules/reagents/chemistry_reagents/other.dm (hunk 5 above)
Comment (Ghommie) [reply code/modules/reagents/chemistry_reagents/other.dm]: Ok, done so, that will alterate balance a little around livers and dylovene, but adjustment can be made later.

Comment (DominikPanic) [reply code/modules/reagents/chemistry_reagents/other.dm]: You don't have to change the typepath, just organize it better

**Pull request merged**